sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -34,13 +35,28 @@ public sealed class LocalWorkspace : Workspace
         private readonly LocalSerializer _serializer = new LocalSerializer();
         private readonly bool _ownsWorkingDir;
         private readonly Task _readyTask;
+        private static readonly SemVersion _minimumVersion = SemVersion.Parse("2.21.0");
 
         /// <inheritdoc/>
         public override string WorkDir { get; }
 
         /// <inheritdoc/>
         public override string? PulumiHome { get; }
 
+        private SemVersion? pulumiVersion;
+        /// <inheritdoc/>
+        public override SemVersion PulumiVersion 
+        { 
+            get 
+            { 
+                if (pulumiVersion == null)
+                {
+                    throw new ArgumentNullException("pulumiVersion");
+                }
+                return pulumiVersion; 
+            }
+        }
```

Comment:
This feels overcomplicated. Would it work if we define the property `PulumiVersion` with a getter and a private setter, initialize it from `GetPulumiVersionAsync`, and get rid of the backing field? We already checked that the version is there in `GetPulumiVersionAsync`, why check it here again?

Reply:
I had it your way when I started, but if I were to do it your way I would have to make `PulumiVersion` nullable, when in reality it will never be null if the `LocalWorkspace` has made it past the constructor. That is the reason for the additional "complication". Is there a less complicated way to do this?

Reply:
I see. You could also do

```cs
public override SemVersion PulumiVersion => pulumiVersion!;
```